File: ios/LivingRoomEdge/LivingRoomEdge/Brain/BrainClient.swift
import Foundation

@MainActor
protocol BrainClient: AnyObject {
    /// **First contact:** Edge → Brain register. Brain returns trusted `edgeId` after confirmation.
    /// Heartbeat must not be called until this succeeds (or a previously assigned edgeId is reused).
    func registerEdge(_ request: EdgeRegisterRequest) async throws -> EdgeRegisterResponse

    /// **Edge → Brain heartbeat** (requires a Brain-issued `info.edgeId`).
    func reportEdgeInfo(_ info: EdgeNodeInfo) async throws
    func fetchPlans(edgeId: String) async throws -> [Plan]
    func report(_ report: ExecutionReport) async throws
}

File: android/living-room-android/src/main/java/com/smarthome/livingroom_android/brain/BrainClient.kt
package com.smarthome.livingroom_android.brain

import com.smarthome.livingroom_android.brain.dto.EdgeNodeInfo
import com.smarthome.livingroom_android.brain.dto.EdgeRegisterRequest
import com.smarthome.livingroom_android.brain.dto.EdgeRegisterResponse
import com.smarthome.livingroom_android.brain.dto.ExecutionReport
import com.smarthome.livingroom_android.brain.dto.Plan

/**
 * Layer 2 brain boundary.
 * Edge → Brain: [registerEdge] then [reportEdgeInfo] (heartbeat).
 * Intent steps are pulled from the business server via HttpCommandSource (not BrainClient).
 */
interface BrainClient {
    suspend fun registerEdge(request: EdgeRegisterRequest): EdgeRegisterResponse

    /**
     * Heartbeat / offline report.
     * @return true if the remote (or mock) heartbeat succeeded.
     */
    suspend fun reportEdgeInfo(info: EdgeNodeInfo): Boolean

    /** Deprecated local plan pull — always empty; kept for interface stability. */
    suspend fun fetchPlans(edgeId: String): List<Plan>

    suspend fun report(report: ExecutionReport)
}
